src/templates/varTemplates.ts

```diff
@@ -1,16 +1,38 @@
 import * as ts from 'typescript'
+import * as vsc from 'vscode'
+import * as _ from 'lodash'
+import { getVariableNameFromCallExpresion } from 'const-name'
 import { CompletionItemBuilder } from '../completionItemBuilder'
 import { BaseExpressionTemplate } from './baseTemplates'
+import { clipByLastIndex } from 'const-name/build/util'
 
 export class VarTemplate extends BaseExpressionTemplate {
   constructor(private keyword: 'var' | 'let' | 'const') {
     super(keyword)
   }
 
   buildCompletionItem(node: ts.Node, indentSize?: number) {
+    let varName: string
+
+    const deriveVariableName = vsc.workspace.getConfiguration('postfix', null).get<boolean>('deriveVariableName')
+    const expr = node.getText();
+    varName = deriveVariableName ?
+      expr.startsWith('new') ?
+        _.lowerFirst(
+          clipByLastIndex(/(.+?)\(/.exec(expr)[1], [' ', '.'])
+        )
+        : getVariableNameFromCallExpresion(expr, {
+          allowApi: true,
+          location: { fileName: '', position: node.pos }
+        })?.replace(/\$/g, '\\$')
+      : undefined;
+
+    if (!varName) {
+      varName = 'name'
+    }

```

Comment:
Well, this is extremely unreadable for me. 
I'm not a fan of too much magic in the code when it's not necessary. 
The more years I am in the business the easier I want the code to look like.

Can you please make it more verbose? For starters I guess this whole part deserves a separate function so that it doesn't clutter main responsibility of `buildCompletionItem`.
I would go with something more explicit like:
```ts
if (/* feature enabled */) {
  if (expr.startsWith('new')) {
     // strategy for types
  } else {
     // strategy for the rest
  }
}

return /* default name */;
```

Reply:
I think we can make thinks a bit easier. Since it have global API I can try to reuse it in this extension: https://github.com/ipatalas/vscode-postfix-ts/pull/65/commits/44449960a2e333013eb4a41b7fefaa46d867f672. Can you please tell wether it still looks messy or not?

Reply:
Hi, sorry, but this is even worse on architectural level. 
This is now very tightly coupled with `const-name` implementation and exposing functionality with global API is not the best idea in my opinion. Polluting global scope (`globalThis.__API_CONST_NAME_VAR_CALL`) in JS and probably all other languages is very rarely a good way to deliver features. Sometimes it can be the only way to solve a problem but that should rather be a last resort when there are no other sensible options which is not the case here. 
This way the code becomes even more confusing because now you have to know it's been magically wired (👉 https://github.com/ipatalas/vscode-postfix-ts/commit/44449960a2e333013eb4a41b7fefaa46d867f672) and this is not simplicity, this is additional complexity.

Another problem I see here is that the lib you provided is responsible for too much and that led to most of those problems. It should be a simple string input, string output function as it doesn't know in what context it's going to be called but it must NOT know that. Libs should be "stupid" in that regard and make no unnecessary assumptions.
Based on the test cases you provided:
![image](https://user-images.githubusercontent.com/2623329/171827425-639cb1b6-95da-4575-8cf9-f3a00e7e3d50.png)
It should only get `Type`, `getSomethingCool` or `usersList` as input and return proper output from that. Only then when it's that simple it can be actually reused in many different contexts (for instance when someone wants to use the same approach in language other than JS/TS).

What to pass to that function is a responsibility of this extension as it knows the context best. We have AST nodes available and this should be used to extract proper part of the expression as this is more reliable than string parsing.
Yes, this will be more lines of code than what you have shown here but it's not always about keeping LoC as low as possible.
Of course this doesn't have to be inline code here in the template builder itself because it's enough responsibility that it can be extracted to a helper method somewhere else (perhaps another util function which can be easily unit-tested and reused in `for` templates for array item name).

It gets little bit more difficult at the beginning because you need to dive in into Typescript AST but it's not that bad when you learn the basics. I encourage you to give it a try. Otherwise I might do that myself next time I have couple of hours to spend on this extension.

Sorry for late reply but didn't have too much time during the week and I wanted to be descriptive about my impressions here and I knew it's gonna take some time to write this comment :)